From 0207e91745de8a3a3811058b88403de171d20464 Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Thu, 30 Jul 2015 08:23:55 +0200
Subject: [PATCH 01/46] Add QDBusConnection argument to AccountsManager

With unit tests we want to fake an AccountsService daemon but of
course we cannot register with the system bus.

Adding a QDBusConnection argument to the constructor (that
defaults to the system bus) allows us to pass a session bus
for unit tests.

Issue: #6
---
 src/accountsservice/accountsmanager.cpp | 8 ++++----
 src/accountsservice/accountsmanager.h   | 3 ++-
 src/accountsservice/accountsmanager_p.h | 2 +-
 3 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/accountsservice/accountsmanager.cpp b/src/accountsservice/accountsmanager.cpp
index e0c32b0..9cb0401 100644
--- a/src/accountsservice/accountsmanager.cpp
+++ b/src/accountsservice/accountsmanager.cpp
@@ -35,12 +35,12 @@ namespace QtAccountsService {
  * AccountsManagerPrivate
  */
 
-AccountsManagerPrivate::AccountsManagerPrivate()
+AccountsManagerPrivate::AccountsManagerPrivate(const QDBusConnection &bus)
 {
     interface = new OrgFreedesktopAccountsInterface(
         QLatin1String("org.freedesktop.Accounts"),
         QLatin1String("/org/freedesktop/Accounts"),
-        QDBusConnection::systemBus());
+        bus);
 }
 
 AccountsManagerPrivate::~AccountsManagerPrivate()
@@ -73,8 +73,8 @@ void AccountsManagerPrivate::_q_userDeleted(const QDBusObjectPath &path)
 /*!
     Constructs a AccountsManager object.
 */
-AccountsManager::AccountsManager()
-    : d_ptr(new AccountsManagerPrivate)
+AccountsManager::AccountsManager(const QDBusConnection &bus)
+    : d_ptr(new AccountsManagerPrivate(bus))
 {
     d_ptr->q_ptr = this;
 
diff --git a/src/accountsservice/accountsmanager.h b/src/accountsservice/accountsmanager.h
index c597b1d..c690764 100644
--- a/src/accountsservice/accountsmanager.h
+++ b/src/accountsservice/accountsmanager.h
@@ -28,6 +28,7 @@
 #define QTACCOUNTSSERVICE_ACCOUNTSMANAGER_H
 
 #include <QtCore/QObject>
+#include <QtDBus/QDBusConnection>
 
 #include <QtAccountsService/UserAccount>
 
@@ -41,7 +42,7 @@ class QTACCOUNTSSERVICE_EXPORT AccountsManager : public QObject
 {
     Q_OBJECT
 public:
-    explicit AccountsManager();
+    explicit AccountsManager(const QDBusConnection &bus = QDBusConnection::systemBus());
     ~AccountsManager();
 
     UserAccount *cacheUser(const QString &userName);
diff --git a/src/accountsservice/accountsmanager_p.h b/src/accountsservice/accountsmanager_p.h
index fd267b3..fdd3b35 100644
--- a/src/accountsservice/accountsmanager_p.h
+++ b/src/accountsservice/accountsmanager_p.h
@@ -50,7 +50,7 @@ class AccountsManagerPrivate
 {
     Q_DECLARE_PUBLIC(AccountsManager)
 public:
-    AccountsManagerPrivate();
+    AccountsManagerPrivate(const QDBusConnection &bus);
     ~AccountsManagerPrivate();
 
     AccountsManager *q_ptr;
-- 
1.9.5

